src/common/hooks/use-setup-tx.ts

```diff
@@ -1,163 +1,41 @@
-      let foundIndex: number | undefined = undefined;
-      wallet.accounts.forEach((account, index) => {
-        const address = getStxAddress({ account, transactionVersion });
-        if (address === payload.stxAddress) {
-          foundIndex = index;
-        }
-      });
-      if (foundIndex !== undefined) {
-        console.debug('switching to index', foundIndex);
-        set(currentAccountIndexStore, foundIndex);
-      } else {
-        console.warn(
-          'No account matches the STX address provided in transaction request:',
-          payload.stxAddress
-        );
-      }
-    },
-    []
-  );
+    if (!requestToken) {
+      handleDecodeRequest();
+    }
+  }, [requestToken, handleDecodeRequest]);
 
   useEffect(() => {
     void handleNetworkSwitch();
     void handleAccountSwitch();
   }, [requestToken, handleNetworkSwitch, handleAccountSwitch]);
 
-  const handlePostConditions = useRecoilCallback(
-    ({ snapshot, set }) => async (currentAddress?: string) => {
-      const payload = await snapshot.getPromise(transactionPayloadStore);
-      if (!payload) return;
-      if (!currentAddress) return;
-      const existingPostConditions = await snapshot.getPromise(postConditionsStore);
-      // change existing post conditions to current account
-      if (existingPostConditions.length) {
-        console.debug('Swapping post conditions address');
-        const newConditions: PostCondition[] = existingPostConditions.map(postCondition => {
-          return {
-            ...postCondition,
-            principal: parsePrincipalString(currentAddress),
-          };
-        });
-        set(postConditionsStore, newConditions);
-        return;
-      } else if (payload.postConditions?.length) {
-        console.debug('Setting up post conditions for transaction request');
-        const newConditions: PostCondition[] = payload.postConditions.map(postCondition => {
-          let payloadCondition: PostCondition;
-          if (typeof postCondition === 'string') {
-            const reader = BufferReader.fromBuffer(Buffer.from(postCondition, 'hex'));
-            payloadCondition = deserializePostCondition(reader);
-          } else {
-            payloadCondition = { ...postCondition };
-          }
-          // override principal with current user
-          const newCondition = {
-            ...payloadCondition,
-            principal: parsePrincipalString(currentAddress),
-          };
-          return newCondition;
-        });
-        set(postConditionsStore, newConditions);
-      }
-    },
-    []
-  );
-
   useEffect(() => {
-    void handlePostConditions(currentAccountStxAddress);
-  }, [requestToken, currentAccountStxAddress, handlePostConditions]);
+    if (currentAccountStxAddress) {
+      if (!previousAccountStxAddress || previousAccountStxAddress !== currentAccountStxAddress) {
+        void handlePostConditions(currentAccountStxAddress);
+      }
+    }
```

Comment:
```suggestion
    if (!currentAccountStxAddress) return;
    if (previousAccountStxAddress || previousAccountStxAddress === currentAccountStxAddress) return
    void handlePostConditions(currentAccountStxAddress);
```
Personal preference is to return early rather than nesting.